deployment/common/changeset/mint_link_timelock.go

```diff
@@ -0,0 +1,50 @@
+package changeset
+
+import (
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+type MintLinkTimelockConfig struct {
+	Amount        *big.Int
+	ChainSelector uint64
+}
+
+var _ deployment.ChangeSet[*MintLinkTimelockConfig] = MintLinkTimelock
+
+// MintLinkTimelock mints LINK to the timelock contract.
+func MintLinkTimelock(e deployment.Environment, req *MintLinkTimelockConfig) (deployment.ChangesetOutput, error) {
+
+	chain := e.Chains[req.ChainSelector]
+	addresses, err := e.ExistingAddresses.AddressesForChain(req.ChainSelector)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	linkState, err := MaybeLoadLinkTokenState(chain, addresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	timelockState, err := MaybeLoadMCMSWithTimelockState(chain, addresses)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	tx, err := linkState.LinkToken.GrantMintRole(chain.DeployerKey, chain.DeployerKey.From)
```

Comment:
I don't think we _ever_ want to grant the deployer key mint rights, big security risk. Would also totally separate the mint/burner management from the act of minting

Reply:
I separated the role granting into a separate changeset, I think it's easier to reason about it that way and allows users to define more granularly who they grant roles to https://github.com/smartcontractkit/chainlink/pull/15512/commits/c8f90363d18d704a669b21bd1817e05ef038e1dc